test/ecto/type_test.exs

```diff
@@ -698,6 +713,14 @@ defmodule Ecto.TypeTest do
       }
       assert Ecto.Type.cast(:utc_datetime_usec, term) == {:ok, @datetime_zero_usec}
 
+      term = %DateTime{
+        calendar: Calendar.ISO, year: 2015, month: 1, day: 24,
+        hour: 9, minute: 50, second: 7, microsecond: {8000, 6},
+        std_offset: 0, utc_offset: 36000,
+        time_zone: "Etc/GMT-10", zone_abbr: "+10"
+      }
+      assert Ecto.Type.cast(:utc_datetime_usec, term) == {:ok, @datetime_usec}
```

Comment:
Both of these added tests basically check the same thing. I was unsure where to add them, as the test description further above says "cast from map", and yet in https://github.com/elixir-ecto/ecto/pull/2850/files#diff-af151a6ff25be431d4b33d60b963bffdR708 we have a DateTime struct. 

I guess either one will do? Maybe keep this one and get rid of the other, as the struct's a bit ugly as a module variable?

Reply:
I personally think it's fine either way and as is too. I slightly lean towards the other example as it's shorter. We could also make the module attribute less ugly by something like this:

```elixir
%{DateTime.from_naive!(~N[2018-12-03 00:00:00.000001], "Etc/UTC") | zone_abbr: "CET", time_zone: "Europe/Berlin", utc_offset: 3600}
```

but on the other hand it's ok to be super explicit in a test.